### PR TITLE
feat(wallet): suggest unlimited allowance as default for swaps  (uplift to 1.36.x)

### PR DESCRIPTION
**Pull request by onyb**
⚠️ QA is necessary, as the changes here include some small bug fixes that were part of a large refactoring from a different PR. Please see the demo in the PR being uplifted.

Uplift of https://github.com/brave/brave-core/pull/12337.

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts
Comment (josheleonard) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts:5]: typo

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts
Comment (josheleonard) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts:325]: maybe re-export `components/brave_wallet_ui/components/extension/sign-panel/style.ts` instead of redefining?

**Review by josheleonard — CHANGES_REQUESTED** [review]

**Comment by onyb** [comment]
@josheleonard The remarks are low in severity, so prefer not to fix them in this PR since it's a cherry-picked commit. Will fix these on Nightly.

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts
Comment (onyb) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts:325]: Yes, we should share the styles for this one. Will do a follow-up on Nightly.

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.36.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/12337#issuecomment-1050504156.

**Pull request merged**
